src/main/java/edu/tamu/iiif/service/MessageListenerService.java

```diff
@@ -0,0 +1,66 @@
+package edu.tamu.iiif.service;
+
+import static edu.tamu.iiif.utility.StringUtility.encode;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import edu.tamu.iiif.controller.ManifestRequest;
+import edu.tamu.iiif.exception.InvalidUrlException;
+import edu.tamu.iiif.exception.UnknownActionException;
+import edu.tamu.iiif.model.RedisManifest;
+import edu.tamu.iiif.model.repo.RedisManifestRepo;
+import edu.tamu.weaver.messaging.annotation.WeaverMessageListener;
+
+@Service
+public class MessageListenerService {
+
+    private static final Logger logger = LoggerFactory.getLogger(MessageListenerService.class);
+
+    @Autowired
+    private RedisManifestRepo manifestRepo;
+
+    @Autowired
+    private List<ManifestService> manifestServices;
+
+    @WeaverMessageListener(destination = "${iiif.messaging.channel}", containerFactory = "topicContainerFactory")
+    private void update(Map<String, String> message) throws UnknownActionException, InvalidUrlException {
+        switch (message.get("action")) {
+        case "METADATA_CREATE":
```

Comment:
We don't not want to use string literals here. Should use a enum from the weaver messaging module.

Reply:
https://github.com/TAMULib/IRIIIFService/commit/136c070c5f2bab08042ec2bdfb6328786a3a1257